packages/ckeditor5-html-support/src/fullpage.ts

```diff
@@ -103,4 +144,81 @@ export default class FullPage extends Plugin {
 			args[ 0 ].trim = false;
 		}, { priority: 'high' } );
 	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public override destroy(): void {
+		super.destroy();
+
+		const allowRenderStylesFromHead = isAllowedRenderStylesFromHead( this.editor );
+
+		if ( allowRenderStylesFromHead ) {
+			this._removeStyleElementsFromDom();
+		}
+	}
+
+	/**
+	 * Checks if in the document exists any `<style>` elements injected by the plugin and removes them,
+	 * so these could be re-rendered later.
+	 * There is used `data-full-page-style-id` attribute to recognize styles injected by the feature.
+	 */
+	private _removeStyleElementsFromDom(): void {
+		const existingStyleElements = Array.from(
+			global.document.querySelectorAll( `[data-full-page-style-id="${ this.editor.id }"]` )
+		);
+
+		for ( const style of existingStyleElements ) {
+			style.remove();
+		}
+	}
+
+	/**
+	 * Extracts `<style>` elements from the full page data and renders them in the main document `<head>`.
+	 * CSS content is sanitized before rendering.
+	 *
+	 * @param fullPageData Represents the full page data passed to the editor as a string.
+	 */
+	private _renderStyleElementsInDom( fullPageData: string ): void {
+		const editor = this.editor;
+
+		// Use DOMParser for easier elements extraction and unwanted code execution prevention.
+		const domParser = editor.data.htmlProcessor.domParser;
+		const doc = domParser.parseFromString( fullPageData, 'text/html' );
```

Comment:
Maybe we could avoid double parsing by extending `HtmlPageDataProcessor` so it could expose a dedicated property for styles?